bundles/mapping/mapmodule/service/map.layer.js

```diff
@@ -688,7 +705,7 @@ Oskari.clazz.define('Oskari.mapframework.service.MapLayerService',
                     layerGroups = group;
                 }
             }
-            return layerGroups || this._layerGroups;
+            return (id) ? layerGroups : layerGroups || this._layerGroups;
```

Comment:
Reading the method implementation if id is falsey layerGroups will always be null so you could just do 

    return (id) ? layerGroups : this._layerGroups;